kata2/schoonology/src/main.rs

```diff
@@ -0,0 +1,178 @@
+extern crate rustbox;
+
+use rustbox::{Color, Key, RustBox};
+use std::error::Error;
+use std::time::{self, Duration};
+
+const STEP_HZ:u64 = 100;
+
+#[derive(Debug, Clone)]
+enum Direction {
+  North,
+  South,
+  East,
+  West,
+}
+
+fn turn_left(dir:Direction) -> Direction {
+  match dir {
+    Direction::North => Direction::West,
+    Direction::South => Direction::East,
+    Direction::East => Direction::North,
+    Direction::West => Direction::South,
+  }
+}
+
+fn turn_right(dir:Direction) -> Direction {
+  match dir {
+    Direction::North => Direction::East,
+    Direction::South => Direction::West,
+    Direction::East => Direction::South,
+    Direction::West => Direction::North,
+  }
+}
+
+#[derive(Debug)]
+struct Ant {
+  x:usize,
+  y:usize,
+  dir:Direction,
+}
+
+struct Game {
+  ant:Ant,
+  grid:Vec<Vec<bool>>,
+}
+
+impl Game {
+  fn new(width:usize, height:usize) -> Game {
+    let mut grid = Vec::with_capacity(width);
+
+    for _ in 0..width {
+      let mut vec = Vec::with_capacity(height);
+      vec.resize(height, false);
+      grid.push(vec);
+    }
+
+    Game{
+      ant:Ant{
+        x:width / 2,
+        y:height / 2,
+        dir:Direction::East,
+      },
+      grid:grid,
+    }
+  }
+
+  fn width(&self) -> usize {
+    self.grid.len()
+  }
+
+  fn height(&self) -> usize {
+    self.grid[0].len()
+  }
+
+  fn turn_ant(&mut self) {
+    if self.grid[self.ant.x][self.ant.y] {
+      self.ant.dir = turn_left(self.ant.dir.clone());
+    } else {
+      self.ant.dir = turn_right(self.ant.dir.clone());
+    }
+  }
+
+  fn flip_tile(&mut self) {
+    self.grid[self.ant.x][self.ant.y] = !self.grid[self.ant.x][self.ant.y];
+  }
+
+  fn move_ant(&mut self) {
+    let width = self.width();
+    let height = self.height();
+
+    match self.ant.dir {
+      Direction::North => self.ant.y = (self.ant.y + height - 1) % height,
+      Direction::South => self.ant.y = (self.ant.y + 1) % height,
+      Direction::East => self.ant.x = (self.ant.x + 1) % width,
+      Direction::West => self.ant.x = (self.ant.x + width - 1) % width,
+    }
+  }
+
+  fn tick(&mut self) {
+    self.turn_ant();
+    self.flip_tile();
+    self.move_ant();
```

Comment:
I really like how simple you've made the rules here. Nice and easy to read, should be easy to change the logic in future.

If you wanted to think about the architecture of the code, I guess you could argue that the game shouldn't know about the inner workings of the ant, and should just tell the ant to act, perhaps even telling the ant what to act on, so that the ant and the game are entirely separate entities. That might make it easier to test and easier to extend in future :)

I hope that's helpful feedback. Overall the code is really easy to read and understand :)